File: src/main/java/com/alealogic/singleproxy/entity/DesktopClient.java
package com.alealogic.singleproxy.entity;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDateTime;

@Data
@Entity
public class DesktopClient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String key;
    private Long customerId;
    private Long sessionId;
    private boolean registered;
    private String version;
    private LocalDateTime lastHeartbeat;
    private String lastIp;
    private String ldflags;
}

File: src/main/java/com/alealogic/singleproxy/entity/DesktopClient.kt
package com.alealogic.singleproxy.entity

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class DesktopClient (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var key: String? = null,
    var customerId: Long? = null,
    var registered: Boolean = false,
    var releaseId: Long? = null,
    var lastHeartbeat: LocalDateTime? = null,
    var lastIp: String? = null,
    var ldflags: String? = null,
    var updateToReleaseId: Long? = null,
    var obsolete: Boolean = false,
    var updateInitiatorDesktopClientId: Long? = null,
    var ipId: String? = null,
    var port: Int? = null,
    var active: Boolean = true
)
